src/Resource/Otp.php

```diff
@@ -52,22 +52,58 @@ public function send(string $to, string $text, string $from = null, string $code
     }
 
     /**
-     * @param string $requestId
+     * Verify an OTP code entered by your customer using request ID received upon sending an OTP
+     *
+     * @param string $requestId Request ID
+     * @param string $code      OTP code
+     *
+     * @return array
+     * @throws AuthenticationException
+     * @throws InvalidPayloadException
+     * @throws ResourceNotFoundException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function verfiyByRequestId(string $requestId, string $code)
+    {
+        $uri = $this->prepareApiUri($this->resourceUri . '/requestid/' . $requestId . '/validate');
+
+        return $this->verify($uri, $code);
+    }
+
+    /**
+     * Verify an OTP code entered by your customer using destination number
+     *
+     * @param string $destination Destination number
      * @param string $code
      *
-     * @return bool|array Returns true if OTP input code is verified successfully otherwise error message
+     * @return array
      * @throws AuthenticationException
      * @throws InvalidPayloadException
      * @throws InvalidResponseException
      * @throws ResourceNotFoundException
      * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    public function verfiyById(string $requestId, string $code)
+    public function verfiyByDestination(string $destination, string $code)
     {
-        $origin = !empty($from) ? $from : '';
+        $uri = $this->prepareApiUri($this->resourceUri . '/' . $destination . '/validate');
```

Comment:
Lets validate the destination number using something like phonenumberutil or other method to ensure they don't pass malicious strings.